twine/cli.py

```diff
@@ -12,10 +12,15 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 import argparse
+import sys
 from typing import Any, List, Tuple
 
-from importlib_metadata import entry_points
-from importlib_metadata import version
+if sys.version_info[:2] >= (3, 10):
+    from importlib.metadata import entry_points
+    from importlib.metadata import version
+else:
+    from importlib_metadata import entry_points
+    from importlib_metadata import version
```

Comment:
I would prefer to do the version check in one place. Given that it's already being done in `twine/__init__.py`, I _think_ this could be something like:

```python
from twine.importlib_metadata import entry_points
from twine.importlib_metadata import version
```

That feels a little exotic, but so does using `importlib_metadata` in the first place (even though it was for a good reason, IIRC).

Reply:
While that's DRY, it also will add an implicit dependency on that import in `twine/__init__.py` and every import in that file slows down the start-up speed of twine. Users often complain of how long CLIs take to start and so I'd rather not start a pattern of importing packages that may have been imported in that file because people tend to glom onto that and re-use that pattern needlessly.

If we want somewhere to centralize imports like this, I'd suggest a separate module altogether, e.g., `twine/_compat.py` where we have this logic and then `twine/__init__.py` can use that and so can other places

Reply:
Since `__init__` already imports `importlib.metadata` for its own use, the argument is only that people might generalize it too much, right?
Should I do this but add a comment explaining it's not a general pattern?

Reply:
Thanks for the explanation, @sigmavirus24. I like the `twine/_compat.py` soluton.

However, after reading through [PR that added standardized on `importlib_metadata`](https://github.com/pypa/twine/pull/732), specifically https://github.com/pypa/twine/pull/732#issuecomment-809755307, I'm still hesitant to accept the complexity this introduces for what feels like a small benefit.

@encukou What do you think about closing this PR, and opening an issue so that references it, so that we can come back to it after the release of 3.10 and/or 3.11?

Reply:
You're free to do that, of course!